vector-im/element-android
# [SDK] Improve location sharing service api (PSF-1004)

## Type of change

- [x] Other : Code refactoring

## Content


All location sharing related APIs have been grouped into the exposed `LocationSharingService` interface.
Unit tests have been added to cover the implementation of this interface.

## Motivation and context


Closes #6156 
Part of #5864 

## Screenshots / GIFs





## Tests



- Enable live location sharing in Settings -> Labs
- Go to a room
- Start sharing location
- Choose to share a static pinned location
- Check it is correctly shared in the room
- Start sharing location
- Choose to share your static location
- Check it is correctly shared in the room
- Start sharing location
- Choose to share your live location
- Check it is correctly shared in the room
- Stop the live location share
- Check it is correctly stopped

## Tested devices

- OS version(s): Android 11

## Checklist
